src/frontends/common/include/openvino/frontend/extension/conversion.hpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/extension.hpp"
+#include "openvino/frontend/node_context.hpp"
+#include "openvino/frontend/visibility.hpp"
+
+namespace ov {
+namespace frontend {
+
+class FRONTEND_API ConversionExtensionBase : public ov::Extension {
+public:
+    using Ptr = std::shared_ptr<ConversionExtensionBase>;
+    explicit ConversionExtensionBase(const std::string& op_type) : m_op_type(op_type) {}
+
+    const std::string& get_op_type() const {
+        return m_op_type;
+    }
+
+    ~ConversionExtensionBase() override = 0;
```

Comment:
It's a little confusing pattern - here the destructor is marked as pure virtual and in the matching cpp file it's defaulted. I think you should just default it here and remove the cpp file since it will be empty afterwards.

Reply:
I haven't decided which method should be pure virtual, this is a way to avoid this
https://stackoverflow.com/a/4641013

I guess most of Base FrontEnd classes should be abstract, no reason to create them itself.

Reply:
@ilyachur what do you think which way is more preferable 
1. pure virtual destructor in Base class with definition in .cpp file
2. protected ctor for Base class

Reply:
Both approaches work, but protected ctos call less questions 😄

Reply:
I decided to remain abstract destructor for a while.